harness/tests/determined/common/experimental/test_trial.py

```diff
@@ -70,3 +70,42 @@ def test_trial_logs_invalid_timestamps(
     with pytest.raises(ValueError):
         list(trialref.logs(timestamp_before=timestamp, timestamp_after=None))
         list(trialref.logs(timestamp_before=None, timestamp_after=timestamp))
+
+
+@pytest.mark.parametrize(
+    "name,valid_timestamp",
+    [
+        ("no_microseconds", "2021-10-26T23:17:12Z"),
+        ("microseconds", "2021-10-26T23:17:12.0000Z"),
+    ],
+)
+@responses.activate
+def test_trial_logs_valid_timestamps(
+    make_trialref: Callable[[int], trial.Trial], name: str, valid_timestamp: str
+) -> None:
+    trialref = make_trialref(1)
+    trial_logs = api_responses.sample_trial_logs()
+    trial_logs_url = f"{_MASTER}/api/v1/trials/{trialref.id}/logs"
+
+    params = {
+        "timestampBefore": valid_timestamp,
+    }
+    responses.get(
+        trial_logs_url,
+        stream=True,
+        body=trial_logs,
+        match=[responses.matchers.query_param_matcher(params, strict_match=False)],
```

Comment:
Using `match` is cute, but I think it could obscure the reason for a possible error. Maybe someone unfamiliar with `responses` will see a `ConnectionError` and think, "Great -- the test failed because the mock library broke. Now what?"

I dunno. It's fine and I know you need to get this out. But maybe for future patches let's just use `matches` to help `responses` deliver different responses based on params.

Reply:
this is a good point, and i think it makes much more sense to not use `match` when the asserts are explicitly checking for request params. i removed it -- thanks for the suggestion.